oak-jcr/src/main/java/org/apache/jackrabbit/oak/jcr/session/SessionContext.java

```diff
@@ -409,6 +421,40 @@ public SecurityProvider getSecurityProvider() {
 
     //-----------------------------------------------------------< internal >---
 
+    private static HttpBlobProvider getOrCreateHttpBlobProvider(Whiteboard wb) {
+        HttpBlobProvider provider = WhiteboardUtils.getService(wb, HttpBlobProvider.class);
```

Comment:
How do you know this is the one from the BlobStore/DataStore used by the NodeStore?

Can this be used to add a `HttpBlobProvider` in an OSGi context at runtime, outside Oak? If yes, is this something that should be allowed?

Reply:
@mreutegg you made this change, do you have any comment?

Reply:
See my reply in [OAK-7570](https://issues.apache.org/jira/browse/OAK-7570?focusedCommentId=16539664&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-16539664).